cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -248,3 +265,131 @@ func (m *GceManager) GetMigNodes(mig *Mig) ([]string, error) {
 	}
 	return result, nil
 }
+
+func (m *GceManager) getMigTemplate(mig *Mig) (*gce.InstanceTemplate, error) {
+	igm, err := m.service.InstanceGroupManagers.Get(mig.Project, mig.Zone, mig.Name).Do()
+	if err != nil {
+		return nil, err
+	}
+	templateUrl, err := url.Parse(igm.InstanceTemplate)
+	if err != nil {
+		return nil, err
+	}
+	_, templateName := path.Split(templateUrl.EscapedPath())
+	instanceTemplate, err := m.service.InstanceTemplates.Get(mig.Project, templateName).Do()
+	if err != nil {
+		return nil, err
+	}
+	return instanceTemplate, nil
+}
+
+func (m *GceManager) buildNodeFromTemplate(mig *Mig, template *gce.InstanceTemplate) (*apiv1.Node, error) {
+
+	if template.Properties == nil {
+		return nil, fmt.Errorf("instance template %s has no properties", template.Name)
+	}
+
+	node := apiv1.Node{}
+	nodeName := fmt.Sprintf("%s-template-%d", template.Name, rand.Int63())
+
+	node.ObjectMeta = metav1.ObjectMeta{
+		Name:     nodeName,
+		SelfLink: fmt.Sprintf("/api/v1/nodes/%s", nodeName),
+		Labels:   map[string]string{},
+	}
+	node.Status = apiv1.NodeStatus{
+		Capacity: apiv1.ResourceList{},
+	}
+	// TODO: get a real value.
+	node.Status.Capacity[apiv1.ResourcePods] = *resource.NewQuantity(110, resource.DecimalSI)
+
+	// TODO: handle custom !!!!
+	// TODO: handle GPU
+	machineType, err := m.service.MachineTypes.Get(mig.Project, mig.Zone, template.Properties.MachineType).Do()
+	if err != nil {
+		return nil, err
+	}
+	node.Status.Capacity[apiv1.ResourceCPU] = *resource.NewQuantity(machineType.GuestCpus, resource.DecimalSI)
+	node.Status.Capacity[apiv1.ResourceMemory] = *resource.NewQuantity(machineType.MemoryMb, resource.DecimalSI)
+	node.Status.Allocatable = node.Status.Capacity
```

Comment:
That is not true in the clusters I run on GCE with kube-up.sh (though it's pretty close). We should at least add a TODO, until we figure it out. Which I think is quite important, as I can imagine overestimating allocatable leading to flapping.